site/public/js/calendar.js

```diff
@@ -0,0 +1,318 @@
+ * @param year : int the current year
+ * @returns {number[]} : array<int> {next_month, year_of_next_month}
+ */
+// eslint-disable-next-line no-unused-vars
+function nextMonth(month, year) {
+    month = month + 1;
+    if (month > 12) {
+        month = month - 12;
+        year = year + 1;
+    }
+    return [month, year];
+}
+
+
+/**
+ * This function creates a Date object based on a string.
+ *
+ * @param datestr : string a string representing a date in the format of YYYY-mm-dd
+ * @returns {Date} a Date object containing the specified date
+ */
+function parseDate(datestr){
+    const temp = datestr.split('-');
+    return new Date(temp[0], temp[1]-1, temp[2]);
+}
+
+/**
+ * This function creates a string in the format of YYYY-mm-dd.
+ *
+ * @param year : int the year
+ * @param month : int the month
+ * @param day : int the date
+ * @returns {string}
+ */
+function dateToStr(year, month, day) {
+    return `${year.toString()}-${month.toString().padStart(2, '0')}-${day.toString().padStart(2, '0')}`;
+}
+
+/**
+ * Creates a HTML table cell that contains a date.
+ *
+ * @param year : int the year of the date
+ * @param month : int the month of the date (1 as January and 12 as December)
+ * @param day : int the date of the date (1 - 31)
+ * @param curr_view_month : int the current month that the calendar is viewing
+ * @param view_semester : if the calendar is viewing the entire semester. If so, the day cell would show both the month and date
+ * @returns {string} the HTML string containing the cell
+ */
+function getDayCell(year, month, day, curr_view_month, view_semester=false) {
+    const cell_date_str = dateToStr(year, month ,day);
+    let content;
+    if (view_semester) {
+        content = `<td class="cal-day-cell cal-cell-expand" id=${cell_date_str}>`;
+    }
+    else {
+        content = `<td class="cal-day-cell" id=${cell_date_str}>`;
+    }
+    content += '<div>';
+    // Title of the day cell
+    content += '<div>';
+    if (view_semester) {
+        content += `<span class="cal-curr-month-date cal-day-title">${monthNamesShort[month]} ${day},</span>`;
+    }
+    else if (month === curr_view_month) {
+        // eslint-disable-next-line no-undef
+        if (day === curr_day && month === curr_month && year === curr_year) {
+            content += `<span class="cal-curr-month-date cal-day-title cal-today-title">${day}</span>`;
+        }
+        else {
+            content += `<span class="cal-curr-month-date cal-day-title">${day}</span>`;
+        }
+    }
+    else {
+        if (month > 12) {
+            month = month % 12;
+        }
+        else if (month <= 0) {
+            month = month + 12;
+        }
+        content += `<span class="cal-next-month-date cal-day-title">${month}/${day}</span>`;
+    }
+    content += '</div>';
+
+    // List all gradeables of other items
+    content += '<div class="cal-cell-items-panel">';
+    // eslint-disable-next-line no-undef
```

Comment:
similar to above, but the inverse, you can use `/* global gradeables_by_date */` at the top of the file to indicate stuff that's coming from outside this script but is defined.